firefox-ios/Client/Frontend/Home/Homepage Rebuild/TopSites/TopSitesManager.swift

```diff
@@ -142,4 +157,28 @@ class TopSitesManager {
         guard !googleTopSite.isEmpty else { return availableSpaceCount }
         return availableSpaceCount - GoogleTopSiteManager.Constants.reservedSpaceCount
     }
+
+    // Keeping the order of the sites, we remove duplicate tiles.
+    // Ex: If a sponsored tile is present then it has precedence over the history sites.
+    // Ex: A default site is present but user has recent history site of the same site.
+    // That recent history tile won't be added.
+    private func removeDuplicates(for sites: [TopSiteState]) -> [TopSiteState] {
```

Comment:
this was copied over from old top sites adaptor, but updated to use `TopSiteState` instead of `Site`.